scripts/gb_model/process_fes_hydrogen_data.py

```diff
@@ -0,0 +1,576 @@
+# SPDX-FileCopyrightText: gb-open-market-model contributors
+#
+# SPDX-License-Identifier: MIT
+
+"""
+Hydrogen data processor.
+
+This script processes hydrogen data from FES workbook.
+"""
+
+import logging
+from pathlib import Path
+
+import numpy as np
+import pandas as pd
+from scipy.interpolate import interp1d
+
+from scripts._helpers import configure_logging, set_scenario_config
+from scripts.gb_model._helpers import _strip_str
+
+logger = logging.getLogger(__name__)
+
+
+def _standardize_year(series: pd.Series) -> pd.Series:
+    """Standardize year format in a pandas Series."""
+    if series.dtype == "object" and "-" in str(series.iloc[0]):
+        series = pd.to_datetime(series).dt.year
+    return series.astype(int) if series.dtype == "object" else series
+
+
+def parse_demand_inputs(
+    demand_sheets: list,
+    demand_sheets_mapping: dict,
+    fes_scenario: str,
+    year_range: list,
+) -> pd.DataFrame:
+    """
+    Parse the hydrogen demand data to the required format.
+
+    Args:
+        demand_sheets (list): List of file paths to demand sheet CSV files
+        demand_sheets_mapping (dict): Nested mapping of fes_year -> sector -> sheet_name
+        fes_scenario (str): FES scenario name to filter by (e.g., "leading the way")
+        year_range (list): Two-element list [start_year, end_year] defining the year range to include
+
+    Returns:
+        df_hydrogen_demand (pd.DataFrame): Processed hydrogen demand data with year as index
+        and sectors as columns. Data values are in MWh for all sectors.
+    """
+
+    # Retrieve annual demand data from sheets
+    demand_data = {}
+    df_hydrogen_demand = pd.DataFrame()
+    # Iterate through the nested structure: year -> sector -> sheet_name
+    for year, sectors in demand_sheets_mapping.items():
+        for sector, sheet_name in sectors.items():
+            # Find the corresponding file in the demand_sheets
+            # Look for files that contain both the year and sheet_name
+            sheet_file = next(
+                f for f in demand_sheets if sheet_name in f and str(year) in f
+            )
+
+            demand_data[sector] = pd.read_csv(sheet_file)
+
+            # Remove whitespace from string columns
+            demand_data[sector] = demand_data[sector].apply(_strip_str)
+
+            # Make all column names lowercase
+            demand_data[sector].columns = demand_data[sector].columns.str.lower()
+
+            # Standardize year format
+            demand_data[sector]["year"] = _standardize_year(demand_data[sector]["year"])
+
+            # Make data column numeric
+            demand_data[sector]["data"] = pd.to_numeric(
+                demand_data[sector]["data"], errors="coerce"
```

Comment:
errors="coerce" risks some data being lost unexpectedly. What does the data that needs to be coerced look like? Could you initially covert these to `nan` and then do `pd.to_numeric`?